packages/kit/src/runtime/client/client.js

```diff
@@ -360,7 +366,12 @@ async function _invalidate() {
 	if (!navigation_result || nav_token !== token) return;
 
 	if (navigation_result.type === 'redirect') {
-		return _goto(new URL(navigation_result.location, current.url).href, {}, 1, nav_token);
+		return _goto(
+			new URL(navigation_result.location, current.url).href,
+			{ replaceState: true },
```

Comment:
I don't think `replaceState: true` should be the default for all redirects coming from invalidations. This should probably be an option if we decide we want this.

Reply:
I think probably it should, because if you do not replace - you become stuck in redirect loop when you try to go back, because user state has changed in such a way, that he or she should be redirected further

For example, login page - after logging in you become "logged in" and you have no business on login page, so it redirects you forward

But still, I understand that this could be considered as a breaking change, so it's your decision